src/mercat/account.rs

```diff
@@ -89,32 +80,27 @@ impl AccountCreatorInitializer for AccountCreator {
                 asset_id,
                 scrt.asset_id_witness.blinding(),
                 generators,
-                valid_asset_ids.as_slice(),
+                valid_asset_ids,
                 BASE,
                 EXPONENT,
             )?,
             rng,
         )?;
 
-        // Gather content and sign it
-        // Account creation is the first transaction. Therefore, nothing has been processed before it.
-        let content = PubAccountContent {
+        Ok(PubAccountTx {
             pub_account: PubAccount {
                 id: account_id,
                 enc_asset_id,
-                memo: AccountMemo::new(scrt.enc_keys.pblc, scrt.sign_keys.public),
+                memo: AccountMemo {
```

Comment:
Please consider moving the `owner_enc_pub_key` out of the memo and then get rid of AccountMemo.